scanner/decisions/0005-scannerv4-read-cvss-data.md

```diff
@@ -0,0 +1,42 @@
+
+## Decision
+
+The updated NVD CVSS GitHub Workflow (CI) has transitioned to sourcing data from the NVD CVE API, rather than the NVD JSON feeds, ensuring the timely update of the NVD CVSS data bundle in Google Storage. To fetch data, the workflow downloads based on both the start index and CVSS v3 severity. For instance, a command listed below could be used: 
+```
+curl "https://services.nvd.nist.gov/rest/json/cves/2.0?cvssV3Severity=CRITICAL&startIndex=0" > critical-1.json
+``` 
+For every severity level, the workflow iteratively exhaust the startIndex to capture all associated data. And there are four severity levels: LOW, MEDIUM, HIGH, CRITICAL. After all json files downloaded, they will be compress to one data bundle as one single compressed file.
+
+We've chosen to base our API requests on CVSS V3 severity to ensure all data integrated into Claircore possesses valid CVSS V3 metrics. Consequently, retrieving data without CVSS V3 information becomes irrelevant.
+
+The CVSS Updater in Central will be modified to download compressed file (generated by CI mentioned above) in the Google bucket. This eliminates the need for JSON parsing in the updater. Scanner V4 will pull the data bundle from Central and populate the Matcher DB with the enrichment data.
+
+A custom enricher is essential for Scanner V4, as it will provide a mechanism to retrieve JSON data from Central, parse the enrichment details, and store them in the Scanner V4 database.  Additionally, our data will be sourced from the NVD CVE API rather than JSON feeds, so the custom enricher can accommodate the distinct data formats. This also offers greater flexibility, allowing us to extract more information from the fetched JSONs than what Claircore offers.
+
+The CVSS Updater in Central will attempt to download the data bundle from the Google bucket using exponential retries in case of partial failures, up to a maximum interval within the next loop, which happens every 4 hours by default.
+
+If the GH workflow (CI) encounters issues accessing or downloading portions of the CVSS data—likely due to NVD API unavailability or rate limiting—it should retry the request after a 6-second delay, capping at three total attempts. If the same HTTP request fails three times consecutively, that specific data section will be skipped. Central retains the data bundle from the prior update. In cases where the current update has data missing but the previous one doesn't (as determined by comparing the sizes of the two data bundles), the earlier data bundle will be employed to support Scanner v4.
```

Comment:
If the GH workflow (CI) encounters issues accessing or downloading portions of the CVSS data—likely due to NVD API unavailability or rate limiting— the same failed http request should be retried in exponential backoff. If the all retries fail, we just abort the whole workflow and wait for next update, which is 4-hour interval.

Reply:
The team should be notified via Slack once the workflow fails (aborted)

Reply:
We need a ticket for sanity check of all bundle in GCS.